### PR TITLE
retry in config, not in env-vars

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: omit
Docs: omit

Some cleaning and refactors after merging: https://github.com/conan-io/conan/pull/5174/files

- Renamed Downloader, Uploader to FileDownloader, FileUploader, so they do not have the same name as the Conan Uploader
- The ``retry`` variable is confusing actually is the total number of attempts. I have added +1 to it by default.
- Passing ``retry`` and ``retry_wait`` as arguments, not env-vars
- Minor pep8 fixes
- The ``tools.download()`` still uses the env-vars, because it does not have access to the config

An alternative solution that seemed cleaner initially was to add ``retry`` and ``retry_wait`` to ``ConanRequester``. It was way cleaner, but then it lost the possibility to replace ConanRequester by ``requests``, forcing some tests (like the RestApiTest) to wrap ``requests`` with a ``ConanRequester``. Would that be ok @lasote? What would you prefer?

@tags: slow

### CODE REVIEW

File: conans/test/unittests/client/rest/uploader_test.py [hunk 1]
```diff
@@ -44,13 +48,15 @@ def put(self, *args, **kwargs):
     def test_403_raises_forbidden_exception_if_token(self):
 
         class MockRequester(object):
+            retry = 0
```
Comment (lasote) [inline conans/test/unittests/client/rest/uploader_test.py]: Is this needed in the current implementation? Are the retry finally members of the requester?

File: conans/test/unittests/client/rest/uploader_test.py [hunk 2]
```diff
@@ -29,13 +31,15 @@ def put(self, *args, **kwargs):
     def test_403_raises_unauthoirzed_exception_if_no_token(self):
 
         class MockRequester(object):
+            retry = 0
```
Comment (lasote) [inline conans/test/unittests/client/rest/uploader_test.py]: Is this needed in the current implementation? Are the retry finally members of the requester?

File: conans/test/unittests/client/rest/uploader_test.py [hunk 3]
```diff
@@ -15,12 +15,14 @@ class UploaderUnitTest(unittest.TestCase):
     def test_401_raises_unauthoirzed_exception(self):
 
         class MockRequester(object):
+            retry = 0
```
Comment (lasote) [inline conans/test/unittests/client/rest/uploader_test.py]: Is this needed in the current implementation? Are the retry finally members of the requester?

**Comment by memsharded** [comment]
Ok, @lasote the implementation in ConanRequester is in this separate PR: https://github.com/conan-io/conan/pull/5244, to be able to compere. If accepted, that would close this one.

**Pull request closed without merging**